### PR TITLE
Add coverage tests for credential templates and vault workflows

### PR DESCRIPTION
**Pull request by ShaojieJiang**
## Summary
- expand backend API tests to exercise credential template edge cases and error responses
- extend vault tests to cover template updates, governance alerts, and sqlite helpers
- broaden OAuth service and workflow model tests to hit remaining branches for full coverage

## Testing
- make lint
- make test

------
https://chatgpt.com/codex/tasks/task_e_68f356dbcf708327bad14d6cbd61356c

### CODE REVIEW

File: src/orcheo/vault/__init__.py [hunk 1]
```diff
@@ -183,6 +426,61 @@ def _load_metadata(
...
+        alert = self._load_alert(alert_id)
+        access_context = context or CredentialAccessContext()
+        unrestricted_context = (
+            access_context.workflow_id is None
+            and access_context.workspace_id is None
+            and not access_context.roles
+        )
+        if (not unrestricted_context) and not alert.scope.allows(access_context):
+            msg = "Governance alert cannot be accessed with the provided context."
+            raise WorkflowScopeError(msg)
```
[the remark this answers is not in the document]
Comment (ShaojieJiang) [reply src/orcheo/vault/__init__.py:477]: @codex, fix this and the eslint errors

**Comment by ShaojieJiang** [comment]
@codex, address the concern above, and fix eslint errors

**Comment by ShaojieJiang** [comment]
@claude, review the PR

**Comment by ShaojieJiang** [comment]
@codex, fix the security issue mentioned by Claude

**Comment by ShaojieJiang** [comment]
@claude, what do you think now?
